swift/ios-shoppe-demo/ModelControllers/OrderManger.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  OrderManger.swift
+//  ios-shoppe-demo
+//
+//  Created on 4/1/20.
+//  Copyright © 2020 FullSTory All rights reserved.
+//
+
+import Foundation
+
+class OrderManager {
+    static var sharedInstance = OrderManager()
+
+    var currentOrder: Order?
+    var itemQuantities = [String: Int]()
+
+    func addProduct(_ product: Product) {
+        currentOrder?.items.append(product)
```

Comment:
This adds a reference to a `Product` from elsewhere in the app, right?  I've seen this lead to unintended side-effects as the object changes elsewhere in the app.  In the Angular app, the object is cloned and then added into the items list.  One other thing to be aware is that `Product` has a `UIImage` member, and that begins to mix the View into the Model.